### ISSUE TITLE
StageProps.MethodOptions: / is being replace by ~1 in ResourcePath

### ISSUE THREAD

**Issue by daniel-michael38** [aws/aws-cdk#15036]
### The Question
I'm utilizing AWS's CDK to develop our API layer. Everything has worked so far and now I am adding in caching. I enabled CacheCluster in the stage and am trying to enable it for a handful of my endpoints. The problem I am facing is the CDK is changing the resource path from something like `/v1/configuration` to `/~1v1~1configuration`. I've tried escaping my /s, but that didn't fix my issue.  I'm also not seeing any documentation on how the string key is suppose to be formatted for `MethodOptions`.

CDK C# Code
```
new Amazon.CDK.AWS.APIGateway.Stage(this, "stagename", new Amazon.CDK.AWS.APIGateway.StageProps
{
    Deployment = deployment,
    StageName = "stagename",
    Variables = stageVariables,
    CacheClusterEnabled = true,
    CacheClusterSize = "1.6",
    DataTraceEnabled = true,
    LoggingLevel = MethodLoggingLevel.INFO,
    MetricsEnabled = true,
    TracingEnabled = true,
    MethodOptions = new Dictionary<string, IMethodDeploymentOptions>
    {
        { "/v1/configuration/GET", new MethodDeploymentOptions { CachingEnabled = true }}
    }
});

```
Resulting Method Settings in the stage:
```
MethodSettings:
  - DataTraceEnabled: true
    HttpMethod: "*"
    LoggingLevel: INFO
    MetricsEnabled: true
    ResourcePath: /*
  - CachingEnabled: true
    HttpMethod: GET
    ResourcePath: /~1v1~1configuration
```
### Environment

  - **CDK CLI Version:** 1.107.0 (build 52c4434)
  - **Node.js Version:** v16.0.0
  - **OS:** MacOS Big Sur 11.3.1
  - **Language (Version):** C#

### Other information
Stackoverflow: https://stackoverflow.com/questions/67892528/aws-cdk-setting-cache-adding-tildes-to-resource-path

**Comment by peterwoodworth** [comment]
Hey @daniel-michael38,

This is actually intended functionality. According to the [cloudformation docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apitgateway-stage-methodsetting.html#cfn-apigateway-stage-methodsetting-resourcepath) `Forward slashes (/) are encoded as ~1 and the initial slash must include a forward slash`

You can see how the CDK does this here:
https://github.com/aws/aws-cdk/blob/52c443474a97a6f8d5cac402e9cb7343f6af9db4/packages/%40aws-cdk/aws-apigateway/lib/util.ts#L14-L43
In short, this will split the path that you submit into different components separated by `/`, then it will pop the last component and use that as the `HttpMethod`. It will then rebuild the string at line 32 by starting it with `/~1` and then combining all the elements of the split string with a `~1` in between.
